### PR TITLE
introduce format.prepareForIsFeed()

### PR DESCRIPTION
**Pull request by staltz**
- [x] Pending on https://github.com/ssb-ngi-pointer/ssb-meta-feeds/pull/48

PR targeting the `partial-replication-changes` branch

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -121,8 +121,12 @@ shows the implementation for 'classic' ed25519 SSB feeds.
 ```js
 {
   name: 'classic',
+  // In case `isFeed` needs to load some state asynchronously
+  prepareForIsFeed(sbot, feedId, cb) {
+    cb()
+  },
   // used in request, block, cleanClock, sbot.post, vectorClock
-  sbotIsFeed(sbot, feedId) {
+  isFeed(sbot, feedId) {
```
Comment (arj03) [inline README.md:129]: ah yes :+1:

File: index.js [hunk 2]
```diff
@@ -196,30 +197,38 @@ exports.init = function (sbot, config) {
...
-      if (blocking) {
-        ebt.block(origFeedId, destFeedId, true)
-      } else if (
-        ebt.state.blocks[origFeedId] &&
-        ebt.state.blocks[origFeedId][destFeedId]
-      ) {
-        // only update unblock if they were already blocked
-        ebt.block(origFeedId, destFeedId, false)
-      }
+      ebt.prepareForIsFeed(() => {
```
Comment (arj03) [inline index.js:218]: this is missing an option?

Comment (staltz) [reply index.js:218]: oH yeAh LOL

**Comment by staltz** [comment]
@arj03 Second attempt now

One thing I was thinking was: should we call `prepareForIsFeed` for every call site of `isFeed`? There are quite many of them

**Comment by arj03** [comment]
Hmm, give me 30 mins to think that over

**Comment by arj03** [comment]
I end up taking back my words but for now I think we should be safe. This is quite tricky to reason about.

We have `initialized` in vectorclock that is used everywhere so that at least makes sure that we have the base state loaded. So the only other case is a race condition where we start doing something before metafeeds has had a chance to register the data, and I think in that regard request is a special case. Take sbot.post which also has isFeed, that one would only be called after you called request on that feed.

**Review by arj03 — APPROVED** [review]

**Pull request merged**